common/djangoapps/django_comment_common/models.py

```diff
@@ -109,6 +109,20 @@ def has_permission(self, permission):
 
         return self.permissions.filter(name=permission).exists()
 
+    @staticmethod
+    def user_has_role_for_course(user, course_id, role_names):
+        """
+        Returns True if the user has one of the given roles for the given course
+        """
+        roles = Role.objects.filter(course_id=course_id, name__in=role_names)
+        for role in roles:
+            users_with_role = role.users.values_list('id', flat=True)
```

Comment:
could we instead do a get or filter to check if a specific user has a record for a particular role?  
it seems inefficient to make a list of every single user with a particular role (maybe i'm wrong)

Reply:
I agree. It seems like we should be able to do something like this:

`return Role.objects.filter(course_id=course_id, name__in=role_names, users=user).exists()`

Reply:
i'll try that thanks